pkg/kubernetes/kubernetes.go

```diff
@@ -405,35 +420,42 @@ func (k *KubeCTLRunner) KubectlExecute(ctx context.Context, args []string, stdin
 		return "", err
 	}
 
-	// if EKS write CA
-	tmpfile, err := os.CreateTemp("", "cluster_*.ca.crt")
-	if err != nil {
-		return "", err
+	kubeCtlOpts := []string{
+		fmt.Sprintf("--server=%s", k.k8sAPIURL),
+		fmt.Sprintf("--token=%s", token.AccessToken),
 	}
-	defer os.Remove(tmpfile.Name())
 
-	_, err = tmpfile.Write(k.clusterCA)
-	if err != nil {
-		return "", err
-	}
+	// Use custom CA if provided
+	if len(k.clusterCA) != 0 {
+		tmpfile, err := os.CreateTemp("", "cluster_*.ca.crt")
+		if err != nil {
+			return "", err
+		}
+		defer os.Remove(tmpfile.Name())
 
-	err = tmpfile.Close()
-	if err != nil {
-		return "", err
+		_, err = tmpfile.Write(k.clusterCA)
+		if err != nil {
+			return "", err
+		}
+
+		err = tmpfile.Close()
+		if err != nil {
+			return "", err
+		}
+
+		kubeCtlOpts = append(
+			kubeCtlOpts,
+			fmt.Sprintf("--certificate-authority=%s", tmpfile.Name()),
+		)
 	}
 
-	args = append([]string{
-		"kubectl",
-		fmt.Sprintf("--server=%s", k.k8sAPIURL),
-		fmt.Sprintf("--token=%s", token.AccessToken),
-		fmt.Sprintf("--certificate-authority=%s", tmpfile.Name()), // TODO: eks
-	}, args...)
+	args = append(kubeCtlOpts, args...)
 	if stdin != "" {
 		args = append(args, "-f", "-")
 	}
 
 	newCommand := func() *exec.Cmd {
-		cmd := exec.Command(args[0], args[1:]...)
+		cmd := exec.Command("kubectl", args[1:]...)
```

Comment:
Can you double check that this is correct? With your changes above, `args[0]` is the first _flag_ instead of the binary name like before, hence this hard-coded value here. But then the next parameter should be `args[0:]...` instead of `args[1:]...`.

Reply:
You are right. args[0] now is the first flag, I tested it in the [Go playground](https://go.dev/play/p/IiNWAyxl8ds). Will update.